lib/GenericsAPI/Utils/MatrixUtil.py

```diff
@@ -2187,24 +2149,35 @@ def perform_variable_stats_matrix(self, params):
 
     def transform_matrix(self, params):
         """
-        transform a matrix
+        Transform a matrix
         """
 
+        OPS = [
+            'abundance_filtering',
+            'relative_abundance',
+            'standardization',
+            'ratio_transformation',
+            'log',
+            'sqrt',
+            'logit',
+        ]
+
         logging.info('Start performing transformation with {}'.format(params))
 
         input_matrix_ref = params.get('input_matrix_ref')
-        workspace_name = params.get('workspace_name')
+        workspace_id = params.get('workspace_id')
         new_matrix_name = params.get('new_matrix_name')
 
-        abundance_filtering_params = params.get('abundance_filtering_params')
+        operations = params.get('operations')
+        abundance_filtering_params = params.get('abundance_filtering_params', {})
         perform_relative_abundance = params.get('perform_relative_abundance', False)
-        standardization_params = params.get('standardization_params')
-        ratio_transformation_params = params.get('ratio_transformation_params')
+        standardization_params = params.get('standardization_params', {})
+        ratio_transformation_params = params.get('ratio_transformation_params', {})
+        log_params = params.get('log_params', {})
 
-        if not isinstance(workspace_name, int):
-            workspace_id = self.dfu.ws_name_to_id(workspace_name)
-        else:
-            workspace_id = workspace_name
+        # validate operations
```

Comment:
should we limit operation size? max 15?

Reply:
Yeah I had a similar thought

Reply:
15 is a good number for now because of the heatmap problem